globus_sdk/transfer/paging.py

```diff
@@ -163,22 +179,68 @@ def iterable_func(self):
 
         This method is the real workhorse of this entire module.
         """
-        # now, cap the limit per request to the max per request size
-        limit = min(self.num_results, self.max_results_per_call)
+        if not self.client_kwargs['params']:
+            self.client_kwargs['params'] = {}
+
+        # to start with, cap the limit per request to the max per request size
+        self.limit = min(self.num_results, self.max_results_per_call)
```

Comment:
Transfer uses a default limit of `100` for `/successful_transfers` so we could consider using that for marker style paging as an optimization that should not put undue burden on Transfer.